core/src/main/scala/tech/beshu/ror/accesscontrol/blocks/rules/FieldsRule.scala

```diff
@@ -45,7 +45,7 @@ class FieldsRule(val settings: Settings)
   override val name: Rule.Name = FieldsRule.name
 
   override def check[B <: BlockContext : BlockContextUpdater](blockContext: B): Task[RuleResult[B]] = Task {
-    if (blockContext.requestContext.isReadOnlyRequest) {
+    if (blockContext.requestContext.isReadOnlyRequest && !blockContext.requestContext.isRorAdminRequest) {
```

Comment:
let's use pattern matching here - it's much more easier to reason about:

```
blockContext.requestContext match {
  case r if r. isRorAdminRequest => RuleResult.Fulfilled(blockContext)
  case r if r. isReadOnlyRequest => handleReadOnlyRequest(blockContext)
  case _ => RuleResult.Rejected()
}
```

As you see it's a little bit different, because we agreed that every ROR action should be matched. Then we should check if it's readonly.